examples/client/index.html

```diff
@@ -53,14 +54,29 @@
 					}
 
 					const query = db.query(queryJson);
-					bindQuery(query);
+
+					let queryHeaders = [];
+					if (queryJson.select) {
+						queryHeaders = queryJson.select;
+					} else {
+						queryHeaders = Array.from(dbFields);
+					}
+
+					bindQuery(query, queryHeaders);
 				};
 
 				document.getElementById("results").innerHTML = "";
 
+				const queryJson = JSON.parse(editor.getValue());
+				if (queryJson.select) {
+					queryHeaders = queryJson.select;
+				} else {
+					queryHeaders = Array.from(dbFields);
+				}
+
 				const query = db.query(JSON.parse(editor.getValue()));
 
-				bindQuery(query);
+				bindQuery(query, queryHeaders);
```

Comment:
I think it would be reasonable to expose a `query.fields()` function that returns the list of selected fields in the API. That would simplify this a bit.

Reply:
I'll address this in a PR that's focused on the API features